apps/src/netsim/PubSubChannel.js

```diff
@@ -61,27 +61,36 @@ PubSubChannel.NullChannel.prototype.unsubscribe = function (eventName, callback)
  * @implements PubSubChannel
  */
 PubSubChannel.PusherChannel = function (pusherApiChannel) {
- /**
-  * The actual Pusher API's channel.
-  * @private {Channel}
-  */
- this.pusherChannel_ = pusherApiChannel;
+  /**
+   * The actual Pusher API's channel.
+   * @private {Channel}
+   */
+  this.pusherChannel_ = pusherApiChannel;
+
+  /**
+   * Cache provided callbacks for easy unsubscribe.
+   * Maps event name to callback.
+   * @type {Object}
+   * @private
+   */
+  this.callbacks_ = {};
 };
 
 /**
  * Subscribe to an event so the given callback is called when the event occurs.
  * @param {string} eventName
- * @param {function{}} callback
+ * @param {function()} callback
  */
 PubSubChannel.PusherChannel.prototype.subscribe = function (eventName, callback) {
   this.pusherChannel_.bind(eventName, callback);
+  this.callbacks_[eventName] = callback;
```

Comment:
Do we want to complain here if this call is going to overwrite the cached callback? Otherwise, we lose the reference to it and can't cleanly unsubscribe.